app/component/itinerary/navigator/hooks/useRealtimeLegs.js

```diff
@@ -197,61 +193,70 @@ const useRealtimeLegs = (relayEnvironment, initialLegs = []) => {
   );
 
   const fetchAndSetRealtimeLegs = useCallback(async () => {
+    const now = Date.now();
     if (
-      !initialLegs?.length ||
-      time >= legTime(initialLegs[initialLegs.length - 1].end)
+      !realTimeLegs?.length ||
+      now >= legTime(realTimeLegs[realTimeLegs.length - 1].end)
     ) {
-      return;
+      setTimeAndRealTimeLegs(prev => ({
+        ...prev,
+        time: now,
+      }));
     }
 
-    const rtLegMap = await queryAndMapRealtimeLegs(planarLegs).catch(err =>
-      // eslint-disable-next-line no-console
-      console.error('Failed to query and map real time legs', err),
+    const rtLegMap = await queryAndMapRealtimeLegs(realTimeLegs, now).catch(
+      err =>
+        // eslint-disable-next-line no-console
+        console.error('Failed to query and map real time legs', err),
     );
 
-    const rtLegs = planarLegs.map(l => {
-      const rtLeg = l.legId ? rtLegMap[l.legId] : null;
-      if (rtLeg) {
-        return {
-          ...l,
-          ...rtLeg,
-          to: {
-            ...l.to,
-            vehicleRentalStation: rtLeg.to.vehicleRentalStation,
-          },
-        };
-      }
-      // copy leg times so that modification will not change original times
-      return { ...l, start: { ...l.start }, end: { ...l.end } };
+    setTimeAndRealTimeLegs(prev => {
+      const rtLegs = prev.realTimeLegs.map(l => {
+        const freezeStart = legTime(l.start) <= now;
+        const freezeEnd = legTime(l.end) <= now;
```

Comment:
I believe this should be done after merging new realtime to previous legs. We should make the freeze based on the very latest informatioin.